core/src/main/java/com/google/errorprone/refaster/PlaceholderUnificationVisitor.java

```diff
@@ -665,7 +665,8 @@ public Boolean visitIdentifier(IdentifierTree node, Unifier unifier) {
     return chooseSubtrees(
         state,
         s -> unifyStatements(node.getStatements(), s),
-        stmts -> maker().Case((JCExpression) node.getExpression(), stmts));
+        s -> unify(node.getBody(), s),
+        (stmts, body) -> maker().Case(CaseTree.CaseKind.STATEMENT, List.<JCExpression>of((JCExpression) node.getExpression()), stmts, body));
```

Comment:
We have to provide 2 additional arguments on JDK 14. It seems like this should be fixed similarly to the `FindIdentifiers` solution (e.g. reflection).